account_payment_sale/models/sale_order.py

```diff
@@ -31,9 +31,12 @@ def _get_payment_mode_vals(self, vals):
                 self.payment_mode_id.bank_account_link == "fixed"
                 and self.payment_mode_id.payment_method_id.code == "manual"
             ):
-                vals[
-                    "partner_bank_id"
-                ] = self.payment_mode_id.fixed_journal_id.bank_account_id.id
+                journal = self.payment_mode_id.fixed_journal_id
+                vals["partner_bank_id"] = journal.bank_account_id.id
+                if journal.payment_debit_account_id:
+                    for line in vals["invoice_line_ids"]:
+                        line[2]["account_id"] = journal.payment_debit_account_id.id
```

Comment:
This is really wrong as this code is supposed to change the account for every lines that would have been prepared when it's executed :warning: 
Fortunately, the invoice lines are prepared after the call to `_prepare_invoice`, therefore this code doesn't mess with the data but it's not doing what it's supposed to do.